ddtelemetry/src/worker/mod.rs

```diff
@@ -474,27 +474,32 @@ impl TelemetryWorker {
     }
 
     // Builds telemetry payloads containing lifecycle events
-    fn build_app_events_batch(&self) -> Vec<Payload> {
+    fn build_app_events_batch(&mut self) -> Vec<Payload> {
         let mut payloads = Vec::new();
 
         if self.data.dependencies.flush_not_empty() {
             payloads.push(data::Payload::AppDependenciesLoaded(
                 data::AppDependenciesLoaded {
-                    dependencies: self.data.dependencies.unflushed().cloned().collect(),
+                    dependencies: self.data.dependencies.drain_unflushed().cloned().collect(),
```

Comment:
Shouldn't these already be removed on a successful send by the call to `removed_flushed`? Yeah the method name has a `d` too much.

Reply:
I've added a new commit that uses the existing method.